hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -201,7 +201,7 @@ class DefaultSource extends RelationProvider
       HadoopFsRelation(
         fileIndex,
         fileIndex.partitionSchema,
-        fileIndex.dataSchema,
+        fileIndex.schema,
```

Comment:
@danny0405 @xiarixiaoyao @xushiyan : Do you folks if this change is ok to do? 
Author has added a comment as to why he made this change. I don't have good context into HoodieFileIndex. would appreciate if you can chime in here.

Reply:
If we make sure that the data files does not contain the partition columns, the original code is actually right. I think we should fix the `HoodieFileIndex.schema` instead for the cases where the partition columns are included in the data files.

Reply:
HoodieFileIndex.dataSchema represents schema without partition fields. while HoodieFileIndex.schema represents hudi table schema (obtained using table schema resolver). Atleast logically it doesn't look right to switch from dataSchema to schema unless we have concrete reasons. 
Also to answer Danny's comment, data files could contain partition columns. Infact, support to remove partition columns from user fields was added only recently. So, for most cases, data files will contain the partition fields/cols.

Reply:
@danny0405 @nsivabalan 
Ok from my understanding 

partiton in data implies dataSchema = schema
for partition not in schema dataSchema is still equal to schema. 

With example say partA is the partition column and is written along with data. so dataSchema is {colA, partA, colB} and schema in commit file will be same {colA, partA, colB}
Now for partition data not written to data file. dataSchema is {colA, colB} and so is commit file schema {colA, colB}
Am i missing something

Reply:
Also there is check which is made in HadoopFSRelation to only add partition column to returned schema if is not there which preserves ordering and doesn't add it in the end. 
Will wait for your comments. Thanks

Reply:
I pushed an update to remove changes in DefaultSource.